csrc/scheduler/normalization_inner_outer.cpp

```diff
@@ -793,1186 +456,220 @@ std::shared_ptr<ReductionParams> innerOuterPersistentHeuristic(
-      normalization_scheduler_utils::getGridOuterNormalizationParams(
-          total_reduction_numel,
-          total_iteration_numel,
-          (int64_t)vectorize_factor,
-          max_persistent_buffer_size);
-
-  NVF_ERROR(outer_params.has_value(), "No valid config found");
-
-  const auto pb_size = outer_params->persistent_buffer_factor;
-  const auto unswitch_factor = outer_params->unswitch_factor;
-
-  auto rparams = std::make_shared<ReductionParams>();
-
-  rparams->persistent_kernel = true;
-  rparams->cross_block_inner_reduction = true;
-  rparams->cross_grid_inner_reduction = true;
-  rparams->grid_dim_iter_dom = ParallelType::BIDx;
-  rparams->grid_dim_inner_reduction = ParallelType::BIDy;
-  rparams->block_dim_inner_reduction = ParallelType::TIDy;
-  rparams->batches_per_block_inner_reduction = pb_size;
-  rparams->multiple_reds_per_blk = true;
-  rparams->vectorize_iter_dom = true;
-  rparams->unroll_factor_iter_dom = (int64_t)vectorize_factor;
-  rparams->block_dim_iter_dom = ParallelType::TIDx;
-  rparams->unroll_factor_inner_reduction = unswitch_factor;
-  rparams->split_grid_dim_iter_dom_inner =
-      ceilDiv(
-          total_iteration_numel / (int64_t)vectorize_factor,
-          outer_params->launch_params.bdimx()) >
-      outer_params->launch_params.gdimx();
-  rparams->compute_persistent_buffer_with_first_consumer = true;
-  rparams->static_bdimx = true;
-  rparams->static_bdimy = true;
-
-  rparams->lparams = LaunchParams(
-      rparams->split_grid_dim_iter_dom_inner
-          ? outer_params->launch_params.gdimx()
-          : LaunchParams::UNINITIALIZED_VAL,
-      LaunchParams::UNINITIALIZED_VAL,
-      LaunchParams::UNINITIALIZED_VAL,
-      outer_params->launch_params.bdimx(),
-      outer_params->launch_params.bdimy(),
-      LaunchParams::UNINITIALIZED_VAL);
-
-  if (isDebugDumpEnabled(DebugDumpOption::SchedulerDebug)) {
-    debug() << "\n===== Reduction Stats ========\n"
-            << "total_reduction_numel: " << total_reduction_numel << "\n"
-            << "total_iteration_numel: " << total_iteration_numel << "\n"
-            << "vectorize_factor: " << vectorize_factor << "\n"
-            << "n_tensor_inputs: " << n_tensor_inputs << "\n"
-            << "max_input_dtype_size: " << max_input_dtype_size << "\n"
-            << "max_persistent_buffer_size: " << max_persistent_buffer_size
-            << "\n"
-            << "persistent_buffer_factor: " << pb_size << "\n"
-            << "block(" << outer_params->launch_params.bdimx() << ", "
-            << outer_params->launch_params.bdimy() << ", 1)" << std::endl;
-    debug() << rparams->toString() << std::endl;
-  }
+    const int64_t estimated_register_count =
+        persistent_buffer_size / bytes_per_register +
+        scheduler_utils::register_overhead;
+    return std::min(
+        estimated_register_count, scheduler_utils::max_registers_per_thread);
+  };
 
-  return rparams;
-}
+  auto getBlocksPerSM = [&](const int64_t threads_per_sm,
```

Comment:
This part shows as changed. Is there any change here? 

Github's diff indicates lots of changes. Maybe it's just false positives? Could you try to minimize the (false?) changes?

Reply:
No change. I tried `hide whitespace` and it removes false changes.
![image](https://github.com/NVIDIA/Fuser/assets/116412316/8ea4f7ae-8ba0-41c5-a291-57e0bc5f446b)